File: TransCoder_tokenized_test_set_functions/cpp/INTEGER_POSITIVE_VALUE_POSITIVE_NEGATIVE_VALUE_ARRAY.cpp
int findInteger ( int arr [ ], int n ) {
  unordered_map < int, int > hash;
  int maximum = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( arr [ i ] < 0 ) hash [ abs ( arr [ i ] ) ] -= 1;
    else hash [ arr [ i ] ] += 1;
  }
  for ( int i = 0;
  i < n;
  i ++ ) if ( hash [ arr [ i ] ] != 0 ) return arr [ i ];
  return - 1;
}

File: TransCoder_tokenized_test_set_functions/java/INTEGER_POSITIVE_VALUE_POSITIVE_NEGATIVE_VALUE_ARRAY.java
static int findInteger ( int arr [ ], int n ) {
  HashMap < Integer, Integer > hash = new HashMap < > ( );
  int maximum = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( arr [ i ] < 0 ) hash . put ( Math . abs ( arr [ i ] ), ( hash . get ( Math . abs ( arr [ i ] ) ) == null ? 0 : hash . get ( Math . abs ( arr [ i ] ) ) ) - 1 );
    else hash . put ( Math . abs ( arr [ i ] ), ( hash . get ( Math . abs ( arr [ i ] ) ) == null ? 0 : hash . get ( Math . abs ( arr [ i ] ) ) ) + 1 );
  }
  for ( int i = 0;
  i < n;
  i ++ ) if ( hash . get ( arr [ i ] ) > 0 ) return arr [ i ];
  return - 1;
}